packages/sources/coingecko/src/endpoint/price.ts

```diff
@@ -21,38 +22,23 @@ const customParams = {
   path: false,
 }
 
-const getCoinId = async (config: Config, symbol: string): Promise<string> => {
-  const url = '/coins/list'
-
-  const options = {
-    ...config.api,
-    url,
-  }
-
-  const response = await Requester.request(options, customError)
-  const coin = response.data.find((x: any) => x.symbol.toLowerCase() === symbol.toLowerCase())
-
-  if (typeof coin === 'undefined') {
-    throw new Error('Coin id not found')
-  }
-
-  return coin.id.toLowerCase()
-}
-
 export const execute: ExecuteWithConfig<Config> = async (request, config) => {
   const validator = new Validator(request, customParams)
   if (validator.error) throw validator.error
 
   const jobRunID = validator.validated.id
-  const symbol = validator.overrideSymbol(AdapterName)
+  const symbol = validator.overrideSymbol(AdapterName) as string
```

Comment:
Shouldn't `overrideSymbol` always return a string?

Reply:
It can also return an array now, because for batched `symbol` inputs they come in as an array. So we may override each one.